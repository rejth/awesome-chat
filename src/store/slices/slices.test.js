import { describe, test, expect } from '@jest/globals';
import chatReducer, {
  setData,
  addChannel,
  removeChannel,
  renameChannel,
  addMessage,
} from './chatSlice.js';

describe('Test chatReducer slice', () => {
  test('setData reducer', () => {
    const state = { data: {} };
    const payload = {};
    expect(chatReducer(state, setData(payload))).toEqual(state);
  });
  test('addChannel reducer', () => {
    const state = { data: { currentChannelId: null, channels: [] } };
    const payload = { id: 1 };
    const result = chatReducer(state, addChannel(payload));
    expect(result.data.channels).toEqual([payload]);
    expect(result.data.currentChannelId).toBe(1);
  });
  test('removeChannel reducer', () => {
    const state = { data: { channels: [{ id: 1 }], messages: [{ channelId: 1 }] } };
    const result = chatReducer(state, removeChannel(1));
    expect(result.data.channels).toEqual([]);
    expect(result.data.messages).toEqual([]);
  });
  test('renameChannel reducer', () => {
    const state = { data: { channels: [{ id: 1, name: 'channel' }] } };
    const payload = { id: 1, name: 'new' };
    const result = chatReducer(state, renameChannel(payload));
    expect(result.data.channels[0]).toEqual(payload);
  });
  test('addMessage reducer', () => {
    const state = { data: { channels: [{ id: 1 }], messages: [] } };
    const payload = { channelId: 1, message: 'hi' };
    const result = chatReducer(state, addMessage(payload));
    expect(result.data.messages[0]).toEqual(payload);
  });
});
