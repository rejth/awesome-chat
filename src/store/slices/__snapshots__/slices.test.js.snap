// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test chatReducer slice addChannel reducer 1`] = `
Array [
  Object {
    "id": 1,
  },
]
`;

exports[`Test chatReducer slice addMessage reducer 1`] = `
Object {
  "channelId": 1,
  "message": "hi",
}
`;

exports[`Test chatReducer slice removeChannel reducer 1`] = `undefined`;

exports[`Test chatReducer slice renameChannel reducer 1`] = `
Object {
  "id": 1,
  "name": "new",
}
`;

exports[`Test chatReducer slice setData reducer 1`] = `
Object {
  "data": Object {},
}
`;
